### ISSUE TITLE
logdump should use subdirs by device name

### ISSUE THREAD

**Issue by jaredmales** [magao-x/MagAOX#161]

**Comment by jaredmales** [comment]
@joseph-long did we reach a consensus on whether we want subdirs by dates on logs?  down to months?  weeks?  days?  We would re-used this for rawimages.

**Comment by joseph-long** [comment]
It seems like the _easiest_ would be by semester, which would roughly match up with how we've organized our backups. But that just makes the need to handle the edge case less frequent, not *impossible*, and it'll probably come up with other applications of the XWCApps. So maybe we should just bite the bullet and use `YYYY/MM/DD/` ?

For the current configuration of length 8192 camwfs xrifs, saving for 12 hours at 2 kHz would be ~10k files per directory, which is well within what the cluster filesystem people think of as a "well-behaved" directory. For on-sky operations we're always spanning a UTC midnight boundary so really it would be half of that.

The downside of having the same scheme for both images and logs is that we'll have a lot of folders with single files in them for the logs. There's also no guarantee that the logs written on YYYY-MM-DD will be in an archive started on YYYY-MM-DD instead of DD-1 or even DD-2, so careful handling is required for trying to do date-related operations (or we just put it all in the database 😎 )

**Comment by jaredmales** [comment]
we can just force a rotate of the logs (and any file) every 24 hours

**Comment by jaredmales** [comment]
and I was thinking about having an option to force the rollover to be at local noon for ground-based applications...

**Comment by joseph-long** [comment]
That sounds like more complexity than it’s worth, if the goal is to identify the log archive for a given utc timestamp before opening it… i like the simplicity of finding doodad_20250101135959999999999.binlog in logs/doodad/2025/01/01/

**Comment by jaredmales** [comment]
What about using julian day number instead of YYYY/MM/DD?  Same number of
directories, but flatter.  Still not easy for a human.

Equivalent would be a directory of 20250101 (or 2025_01_01) instead of
subdirs 2025/01/01.    That has certain efficiencies for globbing, etc.

**Comment by joseph-long** [comment]
Yeah I think daily is fine but I don't think introducing another date system is the move 😅 we already have epoch and UTC and the UTC-to-TAI offset to contend with

**Comment by jaredmales** [comment]
This is coming along: 3aebd06

Going with `device/YYYY_MM_DD/device_YYYYMMDDHHMMSSNNNNNNNNN`

**Comment by jaredmales** [comment]
3d226c3 has subdirs implemented for both logs (and so telems) and images.  

logdump is updated.

@joseph-long what do we need to do for lookyloo?  Anything else?

**Comment by jaredmales** [comment]
we might want to write a utility to convert existing data stores to use this format.  I.e. trawl through logs/ and telems/ and rawimages/ and move stuff around

**Comment by jaredmales** [comment]
implemented
